cupy/_core/internal.pyx

```diff
@@ -53,10 +54,18 @@ cpdef inline tuple get_size(object size):
         )
         return ()
     if cpython.PySequence_Check(size):
-        return tuple(size)
-    if isinstance(size, int):
-        return size,
-    raise ValueError('size should be None, collections.abc.Sequence, or int')
+        try:
+            return tuple(size)
```

Comment:
Should we test each element of `size`?  The fix might be out of scope of the PR, but it might help refactoring.